From 6eb383ba423fa51764b073cad10a5ff45eb1fdc5 Mon Sep 17 00:00:00 2001
From: jgeboski <jgeboski@gmail.com>
Date: Wed, 24 Apr 2013 00:08:46 -0400
Subject: [PATCH] optirun: default to primus then vgl

---
 src/optirun.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/optirun.c b/src/optirun.c
index 65c83d4..a4607ea 100644
--- a/src/optirun.c
+++ b/src/optirun.c
@@ -298,8 +298,8 @@ struct optirun_bridge {
 };
 
 static struct optirun_bridge backends[] = {
-  {"virtualgl", check_virtualgl, run_virtualgl},
   {"primus", check_primus, run_primus},
+  {"virtualgl", check_virtualgl, run_virtualgl},
   {"none", check_none, run_none}, // keep last
   {NULL, NULL, NULL}
 };
-- 
1.7.12.1

